src/dmd/mtype.d

```diff
@@ -7353,3 +7353,49 @@ const(char)* toChars(ScopeRef sr) pure nothrow @nogc @safe
         return names[sr];
     }
 }
+
+/** Returns: `true` iff `t` has any non-`immutable` (mutable or const) indirections.
+ *
+ * In other words, if other references to `t` may be able to change it.
+ * Params:
+ *      t = type to check
+ * Returns:
+ *      true iff `t` contain non-`immutable` indirections.
+ */
+bool hasAliasing(scope Type t)
+{
+    t = t.toBasetype();
+    if (t.isImmutable)
+        return false;
+    else if (auto ts = t.isTypeStruct)
+    {
+        if (auto ti = ts.sym.isInstantiated())
+            if (ti.name.toString == "Rebindable" && // TODO: verify namespace std.typecons?
+                ti.tiargs &&
+                (*ti.tiargs).length == 1)
+                if (auto at = (*ti.tiargs)[0].isType)
+                    return !at.isImmutable;
+        foreach (vd; ts.sym.fields)
+            if (hasAliasing(vd.type))
+                return true;
+    }
+    else if (auto tc = t.isTypeClass)
+        return !tc.isImmutable;
+    else if (auto tp = t.isTypePointer)
+    {
+        if (tp.next.isTypeFunction)
```

Comment:
this is a function call not a member variable. add `()` same for all the others.

Reply:
Why?

Reply:
to disambiguate.

Reply:
Done.